### ISSUE TITLE
Clear addrhint in gossipd?

### ISSUE THREAD

**Issue by ZmnSCPxj** [ElementsProject/lightning#1419]
I wonder, if it would be useful for us to support some way to clear the addrhint that we sent to gossipd?

Prior to #1412 and the #1418, the only way for gossipd to know the address of a node was via addrhint.

However, now we also fall back to DNS seed or (if #1418 is merged) from gossiped `node_announcement`.  But, we only use those if there is no addrhint.

The issue is if the user learns an incorrect IP that does not match the public key.  After failing to connect, the user might decide to try from DNS seed or `node_announcement`.  But currently there is no way to do that except by manual DNS lookup or combing through `listchannels`, and then overwriting the previous addrhint by `connect` command.

**Comment by ZmnSCPxj** [comment]
One way to implement this would be to have two addrhint lists, `maybe_addrhint` and `sure_addrhint`.

The `gossipctl_peer_addrhint` adds to `maybe_addrhint`.

When trying to preform lookup, we get an addrhint from the below places in the below order; the first place we find it is what we try during connect:

1. `maybe_addrhint`
2. `sure_addrhint`
3. DNS lookup
4. routing state table (node announcement gossip)

When we match an addrhint from `maybe_addrhint`, we remove that addrhint from `maybe_addrhint`.  Otherwise we just provide the address.

If we are able to successfully connect out to that peer, we add the address we used as an addrhint to the `sure_addrhint` list.

The effect is below:

1.  If the user provides an address in a `connect` attempt, that address will be used only in that connect attempt.
2.  If the user connects to a brand-new peer without providing an address,  we try to look up by node gossip or DNS seed.
3.  If the user connects to a brand-new peer with an explicit address, but connect fails, and then user tries again without an address, we try to look up by node gossip or DNS seed.
4.  If we previously were connected somehow, then future connect attempts will use the same address as we were able to successfully use before.

Acceptable?
